File: android/src/expo/com/smileidentity/expo/views/SmileIDExpoConsentView.kt
package com.smileidentity.expo.views

import android.content.Context
import androidx.compose.runtime.Composable
import com.smileidentity.SmileID
import com.smileidentity.shared.RNConsent
import com.smileidentity.shared.SmileIDViewConfig
import expo.modules.kotlin.AppContext

/**
 * Expo implementation of Consent using the new architecture
 */
class SmileIDExpoConsentView(
    context: Context,
    appContext: AppContext
) : SmileIDExpoView(context, appContext) {

    var consentInformation: Map<String, Any?>? = null

    override fun renderContent() {
        setContentWithTheme {
            RenderConsentContent()
        }
    }

    @Composable
    private fun RenderConsentContent() {
        val config = SmileIDViewConfig(
            userId = userId,
            jobId = jobId,
            allowAgentMode = allowAgentMode ?: false,
            showInstructions = showInstructions,
            skipApiSubmission = skipApiSubmission,
            showAttribution = showAttribution,
            extraPartnerParams = extraPartnerParams,
            consentInformation = consentInformation
        )

        SmileID.RNConsent(
            config = config,
            onResult = { result ->
                handleResultCallback(result)
            }
        )
    }
}

File: package/ios/expo/views/SmileIDExpoConsentView.swift
import ExpoModulesCore
import UIKit
import SwiftUI
import SmileID

/**
 * Expo implementation of Consent using the new architecture
 */
class SmileIDExpoConsentView: SmileIDExpoView {
    
    var partnerName: String?
    var partnerPrivacyPolicy: String?
    var partnerIcon: String?
    var productName: String?
    
    override func renderContent() {
        let config = createConfig()
        
        // Use SmileIDExtReactNative to create the view
        let uiView = SmileID.RNConsent(
            config: config,
            onResult: { [weak self] result in
                self?.handleResultCallback(result)
            }
        )
        
        // Use the base class helper to set the UIView
        setUIView(uiView)
    }
    
    private func createConfig() -> SmileIDViewConfig {
        var config = SmileIDViewConfig()
        config.partnerName = partnerName
        config.partnerPrivacyPolicy = partnerPrivacyPolicy
        config.logoResName = partnerIcon
        config.productName = productName
        config.showAttribution = showAttribution
        return config
    }
}